### ISSUE TITLE
A simple question on MinkowskiChannelwiseConvolution

### ISSUE THREAD

**Issue by chrockey** [NVIDIA/MinkowskiEngine#351]
Hi, chris

I found that MinkowskiChannelwiseConvolution consumes much more memory than MinkowskiConvolution although it has much less learnable parameters with the same "in_channels" and kernel_size.
I'm not sure what makes MinkowskiChannelwiseConvolution to consume more memory.
I currently suspect that MinkowskiChannelwiseConvolution duplicates input features in (here)[https://github.com/NVIDIA/MinkowskiEngine/blob/c4470e2f0a287fd2cc1f7109c057cad2df00ed3d/MinkowskiEngine/MinkowskiChannelwiseConvolution.py#L186].

Do you have any idea on the reason why MinkowskiChannelwiseConvolution consumes more memory?

Thank you!

**Comment by chrischoy** [comment]
Probably related to how pytorch internally handles the index selection and scalar multiplication.
Writing your own custom CUDA kernel could consume less memory.

**Comment by chrockey** [comment]
I agree with your comments. I'll try to implement a custom CUDA kernel for the channel-wise convolution.
Thanks for your reply!
